pmd-kotlin/src/main/java/net/sourceforge/pmd/cpd/KotlinTokenizer.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd;
+
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+
+import net.sourceforge.pmd.lang.ast.TokenMgrError;
+import net.sourceforge.pmd.lang.kotlin.antlr4.Kotlin;
+
+/**
+ * The Kotlin Tokenizer
+ */
+public class KotlinTokenizer implements Tokenizer {
```

Comment:
this should simply extend `net.sourceforge.pmd.cpd.AntlrTokenizer`, which handles most of the hurdle.

You can check the [`GoTokenizer`](https://github.com/pmd/pmd/blob/master/pmd-go/src/main/java/net/sourceforge/pmd/cpd/GoTokenizer.java) as an example